lib/python/picongpu/input/parameters.py

```diff
@@ -0,0 +1,103 @@
+"""
+This file is part of PIConGPU.
+
+Copyright 2017 PIConGPU contributors
+Authors: Sebastian Starke, Jeffrey Kelling
+License: GPLv3+
+"""
+
+from numpy import log10
+
+
+class Parameter(object):
+    def __init__(self, name, ptype, unit, default, value=None):
+        self.name = name
+        self.type = ptype  # decides if compile time or runtime parameter
+        self.unit = unit
+        self.default = default
+
+        self.value = value if value is not None else default
+
+    def __str__(self):
+        s = ""
+        s += "name=" + self.name
+        s += "\ttype=" + self.type
+        s += "\tunit=" + self.unit
+        s += "\tdefault=" + str(self.default)
+        s += "\tvalue=" + str(self.value)
+        return s
+
+    def as_dict(self):
+        return self.__dict__
+
+    def macro_name(self):
+        return "PARAM_" + self.name.upper()
+
+    def dict_name(self):
+        return self.name.upper()
+
+    def new_with_default(self):
+        return Parameter(self.name, self.type, self.unit, self.default)
+
+
+class UiParameter(Parameter):
+
+    def __init__(self, name, ptype, unit, default, slider_min, slider_max,
+                 slider_step, value=None):
+        Parameter.__init__(self, name, ptype, unit, default, value)
+
+        # for slider widget creation
+        self.min = slider_min
+        self.max = slider_max
+        self.step = slider_step
+
+    def set_value(self, x):
+        """Used as callback function for ipython widget sliders."""
+        self.value = x
+        return self.value
+
+    def get_value(self):
+        return self.value
+
+
+class LinearScaledParameter(UiParameter):
+
+    def __init__(self, name, ptype, unit, default, slider_min, slider_max,
+                 slider_step, scale_factor, value=None):
```

Comment:
if you set `scale_factor = 1.0, ` as the default, you will only need two imports in user code